pkg/otlp/logs/logs_translator.go

```diff
@@ -194,13 +212,15 @@ func flattenAttribute(key string, val pcommon.Value, depth int) map[string]strin
 	return result
 }
 
-func extractHostNameAndServiceName(resourceAttrs pcommon.Map, logAttrs pcommon.Map) (host string, service string) {
-	if src, ok := attributes.SourceFromAttrs(resourceAttrs); ok && src.Kind == source.HostnameKind {
+func (t *Translator) extractHostNameAndServiceName(resourceAttrs pcommon.Map, logAttrs pcommon.Map) (host string, service string) {
+	// TODO: This maps the same resource multiple times and generates the metric multiple times :(
```

Comment:
This should be fixable by refactoring logs code and taking `plog.ResourceLogsSlice` [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/logs_exporter.go#L106) and [here](https://github.com/DataDog/datadog-agent/blob/main/comp/otelcol/otlp/internal/logsagentexporter/logs_exporter.go#L41) as an argument, but yeah don't think this is in the scope of the work here